torch/csrc/jit/ir/ir.h

```diff
@@ -1512,5 +1527,63 @@ struct OperatorSet {
   std::unordered_map<Symbol, std::vector<std::shared_ptr<Operator>>> ops;
 };
 
+template <typename T>
+struct OperatorMap {
+  // Type aliasing
+  using OpMapType = typename std::pair<std::shared_ptr<Operator>, T>;
+  using ValueType = std::vector<OpMapType>;
+  using MapType = std::unordered_map<Symbol, ValueType>;
+
+  OperatorMap() = default;
+  explicit OperatorMap(
+      std::initializer_list<std::pair<std::shared_ptr<Operator>, T>> init) {
+    insert(init);
+  }
+
+  void insert(const std::shared_ptr<Operator>& op, T val) {
+    map[Symbol::fromQualString(op->schema().name())].emplace_back(
+        std::make_pair(op, val));
+  }
+
+  void insert(
+      std::initializer_list<std::pair<std::shared_ptr<Operator>, T>> v) {
+    for (auto& el : v) {
+      insert(el.first, el.second);
+    }
+  }
+
+  void erase_all(const std::shared_ptr<Operator>& op) {
+    map.erase(Symbol::fromQualString(op->schema().name()));
+  }
+
+  bool contains(const Operator& op) const {
```

Comment:
this is checking the QualString but we should also be checking that one of the Operators stored within the vector matches